### PR TITLE
Add support for rules in YAML format

### PR DESCRIPTION
**Pull request by hypnoglow**
## Proposed changes

This commit adds support for defining access rules in YAML format, in
addition to existing JSON format.

YAML is known as a superset of JSON.

This is useful in some cases, where YAML manipulation is preferable over JSON.

## Checklist

- [x] I have read the [contributing guidelines](CONTRIBUTING.md)
- [x] I have read the [security policy](SECURITY.md)
- [x] I confirm that this pull request does not address a security
      vulnerability. If this pull request addresses a security vulnerability, I
      confirm that I got green light (please contact
      [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push
      the changes.
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation within the code base (if appropriate)
- [ ] I have documented my changes in the
      [developer guide](https://github.com/ory/docs) (if appropriate)

### CODE REVIEW

File: rule/fetcher_default_test.go [hunk 1]
```diff
@@ -67,3 +68,58 @@ func TestFetcher(t *testing.T) {
 		})
 	}
 }
+
+func TestFetcherDefault_Fetch_formats(t *testing.T) {
```
Comment (aeneasr) [inline rule/fetcher_default_test.go]: This is a nitpick, we're not using underscores in test names - could you rename this to `TestFetcherDefaultFetchFormats`? :)

**Comment by aeneasr** [comment]
Oh and please run `make format` :)

File: rule/fetcher_default_test.go (hunk 1 above)
Comment (hypnoglow) [reply rule/fetcher_default_test.go]: Oops, I thought you are using "generic" Go convention `TestStruct_Method`. Ok, I will rename this :)

**Comment by hypnoglow** [comment]
> Oh and could you please document this in the oathkeeper docs, preferably in this file.

Sure, I will open a PR for docs soon.

**Comment by hypnoglow** [comment]
@aeneasr Done

**Comment by aeneasr** [comment]
Thank you!

**Pull request merged**